zerver/lib/bugdown/__init__.py

```diff
@@ -961,6 +961,8 @@ def twitter_link(self, url: str) -> Optional[Element]:
 
     def get_url_data(self, e: Element) -> Optional[Tuple[str, str]]:
         if e.tag == "a":
+            if self.markdown.zulip_message is not None:
+                self.markdown.zulip_message.has_link = True
```

Comment:
Can you provide your reasoning for why you expect the changes in this file to cover every link in the message?  

Probably best in a commit message or comment.

(I think you maybe missed my comment that `get_url_data` isn't a good name for a function that sets state?)

Reply:
Agree! And this comment made me realize that I can use the output of `get_url_data` instead of setting the attribute within this function. So I moved this logic to inside the `run` method, where we would expect mutations to happen.